benchmark-ph1ex3/coupled/auxiliary.py

```diff
@@ -0,0 +1,261 @@
+        plt.plot(x, flux, label='F, g={0}'.format(g+1))
+
+    file = 'model1-noTF_axial1_0002.csv'
+    file = pd.read_csv(file)
+    x = np.array(file['y'].tolist())
+    for g in range(G):
+        flux = np.array(file['group'+str(g+1)].tolist())
+        plt.plot(x, flux, label='NF, g={0}'.format(g+1))
+
+    plt.legend(loc="upper left", bbox_to_anchor=(1., 1.), fancybox=True,
+               fontsize=14)
+    plt.xticks(fontsize=16)
+    plt.yticks(fontsize=16)
+    plt.ylabel(r'$\phi \left[\frac{n}{cm^2s}\right]$', fontsize=16)
+    plt.xlabel('z [cm]', fontsize=16)
+    ax.yaxis.get_offset_text().set_fontsize(14)
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+
+    # radial fluxes
+    fig, ax = plt.subplots()
+    save = 'coupledD-3G-radial'
+    file = 'model1_radial1_0002.csv'
+    file = pd.read_csv(file)
+    x = np.array(file['x'].tolist())
+    for g in range(G):
+        flux = np.array(file['group'+str(g+1)].tolist())
+        plt.plot(x, flux, label='F, g={0}'.format(g+1))
+
+    file = 'model1-noTF_radial1_0002.csv'
+    file = pd.read_csv(file)
+    x = np.array(file['x'].tolist())
+    for g in range(G):
+        flux = np.array(file['group'+str(g+1)].tolist())
+        plt.plot(x, flux, label='NF, g={0}'.format(g+1))
+
+    plt.legend(loc="upper left", bbox_to_anchor=(1., 1.), fancybox=True,
+               fontsize=14)
+    plt.xticks(fontsize=16)
+    plt.yticks(fontsize=16)
+    plt.xlabel('r [cm]', fontsize=14)
+    plt.ylabel(r'$\phi \left[\frac{n}{cm^2s}\right]$', fontsize=16)
+    ax.yaxis.get_offset_text().set_fontsize(14)
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+
+    # axial temperature
+    plt.figure()
+    save = 'coupledD-3G-axialT'
+    file = 'model1_axialT_0002.csv'
+    file = pd.read_csv(file)
+    x = np.array(file['y'].tolist())
+    flux = np.array(file['temp'].tolist())
+    plt.plot(x, flux, label='F')
+
+    file = 'model1-noTF_axialT_0002.csv'
+    file = pd.read_csv(file)
+    flux = np.array(file['temp'].tolist())
+    plt.plot(x, flux, label='NF')
+
+    plt.legend(loc="upper right", fontsize=14)
+    plt.xticks(fontsize=14)
+    plt.yticks(fontsize=14)
+    plt.ylabel(r'Temperature [$^{\circ}$C]', fontsize=14)
+    plt.xlabel('z [cm]', fontsize=14)
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+
+    # radial temperature
+    plt.figure()
+    save = 'coupledD-3G-radialT'
+    file = 'model1_radialT_0002.csv'
+    file = pd.read_csv(file)
+    x = np.array(file['x'].tolist())
+    flux = np.array(file['temp'].tolist())
+    plt.plot(x, flux, label='F')
+
+    file = 'model1-noTF_radialT_0002.csv'
+    file = pd.read_csv(file)
+    flux = np.array(file['temp'].tolist())
+    plt.plot(x, flux, label='NF')
+
+    plt.legend(loc="upper right", fontsize=14)
+    plt.xticks(fontsize=14)
+    plt.yticks(fontsize=14)
+    plt.ylabel(r'Temperature [$^{\circ}$C]', fontsize=14)
+    plt.xlabel('r [cm]', fontsize=14)
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+
```

Comment:
```suggestion
    plt.savefig(save, dpi=300, bbox_inches="tight")
    return

```
All functions should have return statements, even if they're just plotting.

Reply:
mmm I don't see why, could you elaborate?
I've done it anyway, I just want to know why it is considered a good practice.

Reply:
^^^ @samgdotson

Reply:
A few reasons: 
1. It's more readable. 
2. Python treats everything as an object. So if you don't have a return statement there is no way to know what "type" of function it is. I.e. if you print a function that has no return statement it will say something like <function object at 230asdfa0s0d9a>. If you print a function that has a return statement it will say something like None or 0 etc. 
3. Due to number 2, it's safer to have a return statement. 

Thanks @katyhuff for clarity